targets/fomu/base.py

```diff
@@ -0,0 +1,451 @@
+        if revision == "pvt" or revision == "evt" or revision == "dvt":
+            self.comb += [
+                If(self.ctrl.storage[3], rgba_pwm[1].eq(self.raw.storage[0])).Else(rgba_pwm[1].eq(ledd_value[0])),
+                If(self.ctrl.storage[4], rgba_pwm[0].eq(self.raw.storage[1])).Else(rgba_pwm[0].eq(ledd_value[1])),
+                If(self.ctrl.storage[5], rgba_pwm[2].eq(self.raw.storage[2])).Else(rgba_pwm[2].eq(ledd_value[2])),
+            ]
+        elif revision == "hacker":
+            self.comb += [
+                If(self.ctrl.storage[3], rgba_pwm[2].eq(self.raw.storage[0])).Else(rgba_pwm[2].eq(ledd_value[0])),
+                If(self.ctrl.storage[4], rgba_pwm[1].eq(self.raw.storage[1])).Else(rgba_pwm[1].eq(ledd_value[1])),
+                If(self.ctrl.storage[5], rgba_pwm[0].eq(self.raw.storage[2])).Else(rgba_pwm[0].eq(ledd_value[2])),
+            ]
+        else:
+            self.comb += [
+                If(self.ctrl.storage[3], rgba_pwm[0].eq(self.raw.storage[0])).Else(rgba_pwm[0].eq(ledd_value[0])),
+                If(self.ctrl.storage[4], rgba_pwm[1].eq(self.raw.storage[1])).Else(rgba_pwm[1].eq(ledd_value[1])),
+                If(self.ctrl.storage[5], rgba_pwm[2].eq(self.raw.storage[2])).Else(rgba_pwm[2].eq(ledd_value[2])),
+            ]
+
+        self.specials += Instance("SB_RGBA_DRV",
+            i_CURREN = self.ctrl.storage[1],
+            i_RGBLEDEN = self.ctrl.storage[2],
+            i_RGB0PWM = rgba_pwm[0],
+            i_RGB1PWM = rgba_pwm[1],
+            i_RGB2PWM = rgba_pwm[2],
+            o_RGB0 = pads.rgb0,
+            o_RGB1 = pads.rgb1,
+            o_RGB2 = pads.rgb2,
+            p_CURRENT_MODE = "0b1",
+            p_RGB0_CURRENT = "0b000011",
+            p_RGB1_CURRENT = "0b000011",
+            p_RGB2_CURRENT = "0b000011",
+        )
+
+        self.specials += Instance("SB_LEDDA_IP",
+            i_LEDDCS = self.dat.re,
+            i_LEDDCLK = ClockSignal(),
+            i_LEDDDAT7 = self.dat.storage[7],
+            i_LEDDDAT6 = self.dat.storage[6],
+            i_LEDDDAT5 = self.dat.storage[5],
+            i_LEDDDAT4 = self.dat.storage[4],
+            i_LEDDDAT3 = self.dat.storage[3],
+            i_LEDDDAT2 = self.dat.storage[2],
+            i_LEDDDAT1 = self.dat.storage[1],
+            i_LEDDDAT0 = self.dat.storage[0],
+            i_LEDDADDR3 = self.addr.storage[3],
+            i_LEDDADDR2 = self.addr.storage[2],
+            i_LEDDADDR1 = self.addr.storage[1],
+            i_LEDDADDR0 = self.addr.storage[0],
+            i_LEDDDEN = self.dat.re,
+            i_LEDDEXE = self.ctrl.storage[0],
+            # o_LEDDON = led_is_on, # Indicates whether LED is on or not
+            # i_LEDDRST = ResetSignal(), # This port doesn't actually exist
+            o_PWMOUT0 = ledd_value[0],
+            o_PWMOUT1 = ledd_value[1],
+            o_PWMOUT2 = ledd_value[2],
+            o_LEDDON = Signal(),
+        )
+
+
+class SBWarmBoot(Module, AutoCSR):
```

Comment:
Should be in https://github.com/timvideos/litex-buildenv/blob/master/gateware/ice40.py